rest-service/manager_rest/security/authentication.py

```diff
@@ -181,7 +181,10 @@ def _authenticate_token(self, token):
         elif not user:
             raise NoAuthProvided()
         else:
-            self._verify_token(token=data[1], user=user)
+            # If we get here, we already validated the new token. Only legacy
+            # tokens need an extra check.
+            if not token.startswith('ctok-'):
+                self._verify_token(token=data[1], user=user)
```

Comment:
Now this makes me a bit uneasy, because _right now_ we indeed have already validated the new token, but I can imagine the possibility of the code above being refactored in such a way, that we get here, and end up not verifying the token in one of the cases.

In other words: I don't like how this tree of elifs is non-complete.

And, we already have an `else` (non-ctok) branch inside of `get_token_status`! Let's just move the `_verify_token` call to there, no?

It's going to be easy to do, except for the `token_verified_cache`. But the new tokens don't use it. So you can either 1) just get rid of the cache (and don't we all love -lines),  2) add a cache to the new tokens as well

Reply:
My intent was to just delete this once we delete old tokens, but we need the stage part to be done before that.